### PR TITLE
gstreamer1-* Update to 1.18.0

### PR DESCRIPTION
**Pull request by Gcenx**
#### Description

Draft update of gstreamer1 packages, it's now using meson build system

All patches have been disabled as they wouldn't apply but were not removed incase these can be rebased

@dbevans as you commit to gstreamer1 packages, @kencu as your a wiz with compiling for older macOS versions.

#### I've tested;
- gstreamer1 +universal -x11
- gstreamer1-gst-plugins-base +universal -x11
- gstreamer1-gst-libav

#### The remaining;
- gstreamer1-gst-plugins-good
- gstreamer1-gst-plugins-bad
- gstreamer1-gst-plugins-ugly

Have not been tested

###### Tested on

macOS 10.14.6 18G6032
Xcode 11.3.1 11C505

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by dbevans** [comment]
Testing now.  

Minor nit: the commit message (not just the PR title) needs to be revised to indicate that all the gstreamer1 ports are being updated to version 1.18.0.

Will respond further after testing is complete. Thanks for the PR.

**Comment by Gcenx** [comment]
Ah yes, I also didn’t indicate the ticket this closes.

**Review by dbevans — CHANGES_REQUESTED** [review]
Some issues to start:

Dependencies:
* gobject-introspection needs to be a library dependency because it provides basic .gir files and .typelibs as well as various libraries that ports using introspection link against.
* same for glib2
* I'm suspicious about m4 since that is autotools specific
* bison and flex probably ought to be build dependencies since they are usually used only at build

Configuring python: 
* I don't like the. idea of using symbolic links to python3.  Better IMO is to patch meson.build with the actual path to our python binary.  See other meson ports for examples.

Test failure:

Out of the box gstreamer1 failed during configuration as follows:

```
:info:configure Executing:  cd "/opt/local/var/macports/build/_Volumes_GitRepos_github.com_macports_macports-ports_gnome_gstreamer1/gstreamer1/work/gstreamer-1.18.0" && /opt/local/bin/meson --prefix=/opt/local --disable-examples --disable-fatal-warnings --disable-silent-rules --without-dw --without-unwind /opt/local/var/macports/build/_Volumes_GitRepos_github.com_macports_macports-ports_gnome_gstreamer1/gstreamer1/work/gstreamer-1.18.0 /opt/local/var/macports/build/_Volumes_GitRepos_github.com_macports_macports-ports_gnome_gstreamer1/gstreamer1/work/build
:debug:configure system:  cd "/opt/local/var/macports/build/_Volumes_GitRepos_github.com_macports_macports-ports_gnome_gstreamer1/gstreamer1/work/gstreamer-1.18.0" && /opt/local/bin/meson --prefix=/opt/local --disable-examples --disable-fatal-warnings --disable-silent-rules --without-dw --without-unwind /opt/local/var/macports/build/_Volumes_GitRepos_github.com_macports_macports-ports_gnome_gstreamer1/gstreamer1/work/gstreamer-1.18.0 /opt/local/var/macports/build/_Volumes_GitRepos_github.com_macports_macports-ports_gnome_gstreamer1/gstreamer1/work/build
:info:configure usage: meson [-h] {setup,configure,dist,install,introspect,init,test,wrap,subprojects,help,rewrite,compile} ...
:info:configure meson: error: unrecognized arguments: --disable-examples --disable-fatal-warnings --disable-silent-rules --without-dw --without-unwind
:info:configure Command failed:  cd "/opt/local/var/macports/build/_Volumes_GitRepos_github.com_macports_macports-ports_gnome_gstreamer1/gstreamer1/work/gstreamer-1.18.0" && /opt/local/bin/meson --prefix=/opt/local --disable-examples --disable-fatal-warnings --disable-silent-rules --without-dw --without-unwind /opt/local/var/macports/build/_Volumes_GitRepos_github.com_macports_macports-ports_gnome_gstreamer1/gstreamer1/work/gstreamer-1.18.0 /opt/local/var/macports/build/_Volumes_GitRepos_github.com_macports_macports-ports_gnome_gstreamer1/gstreamer1/work/build
:info:configure Exit code: 2
:error:configure Failed to configure gstreamer1: configure failure: command execution failed

```

Be sure to remove all lingering autotools style configure options and replace them with meson-style options such as -Doption=value if needed.  See meson_options.txt for the list of available options.

**Comment by Gcenx** [comment]
That’s strange gstreamer1 and base is installed locally, I’ll check that copy then commit the changes.

The portfiles that used the gobject-introspection portgroup should have this added already (see above)

I’ll remove m4 and add glib2.

The reason for the python3 symlink was some additional utility’s also needed python3 and failed once patched

**Comment by Gcenx** [comment]
The python3 symlinking still happens currently but `gstreamer1` & `gstreamer1-gst-plugins-base` now compile.

I built using `gstreamer1-gst-plugins-base +universal -x11`, as I don't like having X11 libraries installed into macports

File: gnome/gstreamer1-gst-libav/Portfile [hunk 1]
```diff
@@ -22,17 +23,24 @@ master_sites        https://gstreamer.freedesktop.org/src/${my_name}/
...
+depends_lib-append          port:gobject-introspection \
+                            port:gstreamer1-gst-plugins-base \
+                            port:ffmpeg
 
-# This will cause the installed ffmpeg headers to be used, remove it.
-configure.cppflags-delete -I${prefix}/include
+# meson.build uses python3 for multiple checks
+pre-configure {
+    file mkdir ${workpath}/bin
+    file link -symbolic ${workpath}/bin/python3 ${prefix}/bin/python3.8
```
Comment (ryandesign) [inline gnome/gstreamer1-gst-libav/Portfile:41]: Note that MacPorts has an `ln` procedure that works like the `ln` executable so you could write:
```
ln -s ${prefix}/bin/python3.8 ${workpath}/bin/python3
```

Comment (Gcenx) [reply gnome/gstreamer1-gst-libav/Portfile]: I wasn't sure the more "macports" correct way as some Portfiles did this some used ln, I'll swap this over to using `ln` and update the comment.

**Comment by Gcenx** [comment]
Setting `-Dintrospection=enabled` is causing linker errors due to undefined symbols, removing the option gstreamer1 builds without issue.

Does this option need to be enabled?

**Comment by kencu** [comment]
@dbevans  -- would you like this PR, or shall we close it and wait for your update?

**Comment by Gcenx** [comment]
@kencu I’m thinking of just closing this, maybe I’ll get around to updating the Macports specific patches eventually or someone else will.

I’ll be carrying gstreamer1.18.3 Portfiles locally until this happens.

**Comment by kencu** [comment]
Dave is a very dedicated MacPorts maintainer, going back years and years. He doesn't usually do his updates via PRs or tickets, and he updates things regularly, but on his own timelines. We have all pretty much decided to stop doing any PRs for ports he maintains, as it's just not helpful to him. So -- probably that's the best idea.

**Comment by mascguy** [comment]
FYI, I'm in the process of updating these ports, and just stumbled upon this PR (and the associated ticket). And this will definitely save me some time.

Thanks Dean!

**Comment by Gcenx** [comment]
@mascguy This is rather old I have a more complete versions on https://github.com/Gcenx/macports-gstreamer1

Some option will still lightly need tweaking as the newer revision we’ve worked on is only aimed at provided codecs for using by wine.

**Pull request closed without merging**